### PR TITLE
Add minimum disk space requirements for the components

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v6.1 (TiDB 6.1 versions)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/9202
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: hardware-and-software-requirements.md [hunk 1]
```diff
@@ -127,6 +127,17 @@ As an open source distributed NewSQL database, TiDB requires the following netwo
...
+## Disk space requirements
+
+| Component | Disk space requirement |
+| :-- | :-- |
+| TiDB |  |
+| PD | Reserve at least 30 GB for data and log disks |
+| TiKV | Reserve at least 100 GB for data and log disks <br/> Ensure that the disk usage is below 90% |
+| TiFlash | TiFlash binary: Occupies 2.1 GB. Considering log space, you are advised to reserve no less than 30 GB space for the deployment directory <br/> Data disk: No less than 100 GB is recommended. 20 GB is the minimum requirement. Ensure that the disk usage is below 80% |
+| TiUP | A TiUP cluster occupies hundreds of megabytes. During deployment, the TiUP cluster requires no more than 1 MB /tmp space to restore temporary configuration files |
+ Ngmonitoring | Conprof: 3 x 1 GB x Number of components + 20 GB (reserved space. 3 days, 1 GB for each component per day)<br/>TopSQL: 30 x 50 MB x Number of components (30 days, 50 MB for each component per day, shared with Continuous Profiling) |
```
Comment (JaySon-Huang) [inline hardware-and-software-requirements.md:139]:
```suggestion
| Ngmonitoring | Conprof: 3 x 1 GB x Number of components + 20 GB (reserved space. 3 days, 1 GB for each component per day)<br/>TopSQL: 30 x 50 MB x Number of components (30 days, 50 MB for each component per day, shared with Continuous Profiling) |
```

File: hardware-and-software-requirements.md [hunk 2]
```diff
@@ -127,6 +127,17 @@ As an open source distributed NewSQL database, TiDB requires the following netwo
...
 | Alertmanager | 9094 | the alert communication port |
 
+## Disk space requirements
+
+| Component | Disk space requirement |
+| :-- | :-- |
+| TiDB |  |
+| PD | Reserve at least 30 GB for data and log disks |
+| TiKV | Reserve at least 100 GB for data and log disks <br/> Ensure that the disk usage is below 90% |
+| TiFlash | TiFlash binary: Occupies 2.1 GB. Considering log space, you are advised to reserve no less than 30 GB space for the deployment directory <br/> Data disk: No less than 100 GB is recommended. 20 GB is the minimum requirement. Ensure that the disk usage is below 80% |
```
Comment (shichun-0415) [inline hardware-and-software-requirements.md:137]:
```suggestion
| TiFlash | Log disk: Reserve at least 30 GB <br/> Data disk: Reserve at least 100 GB. Ensure that the disk usage is below 80% |
```

File: hardware-and-software-requirements.md [hunk 3]
```diff
@@ -127,6 +127,17 @@ As an open source distributed NewSQL database, TiDB requires the following netwo
 | Alertmanager | 9093 | the port for the alert web service |
 | Alertmanager | 9094 | the alert communication port |
 
+## Disk space requirements
+
+| Component | Disk space requirement |
+| :-- | :-- |
+| TiDB |  |
+| PD | Reserve at least 30 GB for data and log disks |
```
Comment (shichun-0415) [inline hardware-and-software-requirements.md:135]:
```suggestion
| PD | Reserve at least 20 GB for data and log disks |
```

File: hardware-and-software-requirements.md [hunk 4]
```diff
@@ -127,6 +127,17 @@ As an open source distributed NewSQL database, TiDB requires the following netwo
...
 | Alertmanager | 9094 | the alert communication port |
 
+## Disk space requirements
+
+| Component | Disk space requirement | Healthy disk usage |
+| :-- | :-- | :-- |
+| TiDB | Reserve at least 30 GB for the log disk | Lower than 90% |
+| PD | Reserve at least 20 GB for the data disk and the log disk, respectively | Lower than 90% |
+| TiKV | Reserve at least 100 GB for the data disk and the log disk, respectively | Lower than 80% |
+| TiFlash | Reserve at least 100 GB and 30 GB for the data disk and log disk, respectively | Lower than 80% |
```
Comment (JaySon-Huang) [inline hardware-and-software-requirements.md:137]:
```suggestion
| TiFlash | Reserve at least 100 GB and 30 GB for the data disk and the log disk, respectively | Lower than 80% |
```
Should there be a "the" before "log disk"?

**Review by JaySon-Huang — APPROVED** [review]
LGTM

File: hardware-and-software-requirements.md [hunk 5]
```diff
@@ -127,6 +127,17 @@ As an open source distributed NewSQL database, TiDB requires the following netwo
 | Alertmanager | 9093 | the port for the alert web service |
 | Alertmanager | 9094 | the alert communication port |
 
+## Disk space requirements
+
+| Component | Disk space requirement | Healthy disk usage |
+| :-- | :-- | :-- |
+| TiDB | Reserve at least 30 GB for the log disk | Lower than 90% |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:134]: I don't think "Reserve" is necessary here.
```suggestion
| TiDB | At least 30 GB for the log disk | Lower than 90% |
```

File: hardware-and-software-requirements.md [hunk 6]
```diff
@@ -127,6 +127,17 @@ As an open source distributed NewSQL database, TiDB requires the following netwo
...
+| PD | Reserve at least 20 GB for the data disk and the log disk, respectively | Lower than 90% |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:135]:
```suggestion
| PD | At least 20 GB for the data disk and for the log disk, respectively | Lower than 90% |
```

File: hardware-and-software-requirements.md [hunk 7]
```diff
@@ -127,6 +127,17 @@ As an open source distributed NewSQL database, TiDB requires the following netwo
...
+| TiKV | Reserve at least 100 GB for the data disk and the log disk, respectively | Lower than 80% |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:136]: ditto

File: hardware-and-software-requirements.md [hunk 8]
```diff
@@ -127,6 +127,17 @@ As an open source distributed NewSQL database, TiDB requires the following netwo
...
+| TiFlash | Reserve at least 100 GB and 30 GB for the data disk and the log disk, respectively | Lower than 80% |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:137]:
```suggestion
| TiFlash | At least 100 GB for the data disk, and at least 30 GB for the log disk, respectively | Lower than 80% |
```

File: hardware-and-software-requirements.md [hunk 9]
```diff
@@ -127,6 +127,17 @@ As an open source distributed NewSQL database, TiDB requires the following netwo
...
+| TiUP | A TiUP cluster occupies no more than 1 GB. During deployment, the TiUP cluster requires no more than 1 MB /tmp space to restore temporary configuration files | N/A |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:138]: 见 https://github.com/pingcap/docs-cn/pull/9202/files#r867659972
```suggestion
| TiUP | A single TiUP cluster occupies no more than 1 GB. During the cluster deployment process, the TiUP cluster requires no more than 1 MB /tmp space to restore temporary configuration files. | N/A |
```

File: hardware-and-software-requirements.md [hunk 10]
```diff
@@ -127,6 +127,17 @@ As an open source distributed NewSQL database, TiDB requires the following netwo
...
+| Ngmonitoring | Conprof: 3 x 1 GB x Number of components (3 days, each component occupies about 1 GB per day) + 20 GB reserved space <br/> Top SQL: 30 x 50 MB x Number of components (30 days, each component occupies about 50 MB per day) <br/> Conprof and Top SQL share the reserved space | N/A |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:139]:
```suggestion
| Ngmonitoring | Conprof: 3 x 1 GB x Number of components (which means each component occupies about 1 GB per day in a total of 3 days) + 20 GB reserved space <br/> Top SQL: 30 x 50 MB x Number of components (which means each component occupies about 50 MB per day in a total of 30 days) <br/> Conprof and Top SQL share the reserved space | N/A |
```

File: hardware-and-software-requirements.md [hunk 11]
```diff
@@ -127,6 +127,17 @@ As an open source distributed NewSQL database, TiDB requires the following netwo
...
 
+## Disk space requirements
+
+| Component | Disk space requirement | Healthy disk usage |
+| :-- | :-- | :-- |
+| TiDB | At least 30 GB for the log disk | Lower than 90% |
+| PD | At least 20 GB for the data disk and for the log disk, respectively | Lower than 90% |
+| TiKV | At least 100 GB for the data disk and for the log disk, respectively | Lower than 80% |
+| TiFlash | At least 100 GB for the data disk and at least 30 GB for the log disk, respectively | Lower than 80% |
+| TiUP | <ul><li>Control machine: No more than 1 GB space is required for deploying a single version of TiDB cluster. The space required increases if multiple versions of TiDB cluster are deployed. </li> <li> Deployment servers (machines where the TiDB components run): TiFlash occupies about 700 MB space and other components (such as PD, TiDB, and TiKV) occupy about 200 MB space respectively. During the cluster deployment process, the TiUP cluster requires less than 1 MB temporary space (`/tmp` directory) to store temporary files.</li></ul>| N/A |
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:138]: The Chinese version says "其他组件（PD、TiDB、TiKV 等）占用约 200 MB 空间" WITHOUT mentioning "respectively". This is inconsistent.
```suggestion
| TiUP | <ul><li>Control machine: No more than 1 GB space is required for deploying a TiDB cluster of a single version. The space required increases if TiDB clusters of multiple versions are deployed. </li> <li> Deployment servers (machines where the TiDB components run): TiFlash occupies about 700 MB space and other components (such as PD, TiDB, and TiKV) occupy about 200 MB space respectively. During the cluster deployment process, the TiUP cluster requires less than 1 MB of temporary space (`/tmp` directory) to store temporary files.</li></ul>| N/A |
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**